src/application/services/native_spec_service.py

```diff
@@ -1,5 +1,6 @@
 """Generic native spec processing service."""
 
+import logging
```

Comment:
So, if you look at other services, you will see we pass the logger port in the constructor - we should probably do the same here.